rustls/src/quic.rs

```diff
@@ -1039,7 +1039,70 @@ mod tests {
     use std::prelude::v1::*;
 
     use super::PacketKey;
-    use crate::quic::HeaderProtectionKey;
+    use crate::crypto::tls13::OkmBlock;
+    use crate::enums::CipherSuite;
+    use crate::quic::{HeaderProtectionKey, Secrets, Version};
+    use crate::{Side, TEST_PROVIDERS};
+
+    #[test]
+    fn key_update_test_vector() {
+        fn equal_okm(x: &OkmBlock, y: &OkmBlock) -> bool {
+            x.as_ref() == y.as_ref()
+        }
+
+        for provider in TEST_PROVIDERS {
+            let aes_128_gcm = provider
+                .tls13_cipher_suites
+                .iter()
+                .find(|cs| cs.common.suite == CipherSuite::TLS13_AES_128_GCM_SHA256)
+                .unwrap();
+            let quic_suite = aes_128_gcm.quic.unwrap();
+
+            let mut secrets = Secrets::new(
```

Comment:
`Secrets::new()` is private, so was trying to keep this outside of provider-specific tests.